src/megaclient.cpp

```diff
@@ -12971,6 +12972,177 @@ node_vector *MegaClient::nodesbyfingerprint(FileFingerprint* fingerprint)
+    MegaClient* mc;
+
+    const static string webclient_is_image_def = ".jpg.jpeg.gif.bmp.png.";
+    const static string webclient_is_image_raw = ".3fr.arw.cr2.crw.ciff.cs1.dcr.dng.erf.iiq.k25.kdc.mef.mos.mrw.nef.nrw.orf.pef.raf.raw.rw2.rwl.sr2.srf.srw.x3f.";
+    const static string webclient_is_image_thumb = "psd.svg.tif.tiff.webp";  // leaving out .pdf
+    const static string webclient_mime_photo_extensions = ".3ds.bmp.btif.cgm.cmx.djv.djvu.dwg.dxf.fbs.fh.fh4.fh5.fh7.fhc.fpx.fst.g3.gif.heic.heif.ico.ief.jpe.jpeg.jpg.ktx.mdi.mmr.npx.pbm.pct.pcx.pgm.pic.png.pnm.ppm.psd.ras.rgb.rlc.sgi.sid.svg.svgz.tga.tif.tiff.uvg.uvi.uvvg.uvvi.wbmp.wdp.webp.xbm.xif.xpm.xwd.";
+    const static string webclient_mime_video_extensions = ".3g2.3gp.asf.asx.avi.dvb.f4v.fli.flv.fvt.h261.h263.h264.jpgm.jpgv.jpm.m1v.m2v.m4u.m4v.mj2.mjp2.mk3d.mks.mkv.mng.mov.movie.mp4.mp4v.mpe.mpeg.mpg.mpg4.mxu.ogv.pyv.qt.smv.uvh.uvm.uvp.uvs.uvu.uvv.uvvh.uvvm.uvvp.uvvs.uvvu.uvvv.viv.vob.webm.wm.wmv.wmx.wvx.";
+
+    static bool isvideo(const Node* n, char ext[10])
+    {
+        if (n->hasfileattribute(fa_media) && n->nodekey.size() == FILENODEKEYLENGTH)
+        {
+#ifdef USE_MEDIAINFO
+            if (mc->mediaFileInfo.mediaCodecsReceived)
+            {
+                MediaProperties mp = MediaProperties::decodeMediaPropertiesAttributes(n->fileattrstring, (uint32_t*)(n->nodekey.data() + FILENODEKEYLENGTH / 2));
+                unsigned videocodec = mp.videocodecid;
+                if (!videocodec && mp.shortformat) {
+                    auto& v =mc->mediaFileInfo.mediaCodecs.shortformats;
+                    if (mp.shortformat < v.size())
+                    {
+                        videocodec = v[mp.shortformat].videocodecid;
+                    }
+                }
+                // approximation: the webclient has a lot of logic to determine if a particular codec is playable in that browser.  We'll just base our decision on the presence of a video codec.
+                if (!videocodec)
+                {
+                    return false; // otherwise double-check by extension
+                }
+            }
+#endif  
+        }
+        return webclient_mime_video_extensions.find(ext) != string::npos;
+    }
+
+    static bool ismedia(const Node* n)
+    {
+        // evaluate according to the webclient rules, so that we get exactly the same bucketing.
+        string localname, name = n->displayname();
+        mc->fsaccess->path2local(&name, &localname);
+        char ext[10];
+        if (mc->fsaccess->getextension(&localname, ext, sizeof(ext)))
+        {
+            strcat(ext, ".");
+
+            if (webclient_is_image_def.find(ext) != string::npos ||
+                webclient_is_image_raw.find(ext) != string::npos ||
+                (webclient_mime_photo_extensions.find(ext) != string::npos && n->hasfileattribute(GfxProc::PREVIEW)) ||
```

Comment:
Why do we skip photos without a preview?

Reply:
The reason that's in there is that it's imitating the logic in the webclient - if you search that codebase for `is_image3` you'll see it makes the same test.  Probably due to some case where a common enough file extension is something other than a photo, maybe some program's proprietary format.   The most common photo extensions are covered by the first two tests anyway, so we will mostly not depend on that.  cheers